src/cascadia/TerminalCore/ICoreSettings.idl

```diff
@@ -3,6 +3,24 @@
 
 namespace Microsoft.Terminal.Core
 {
+    // TerminalCore declares its own Color struct to avoid depending
+    // on Windows.UI.Color and to avoid passing around unclothed uint32s.
+    // It is supported by til::color for conversions in and out of WinRT land.
+    struct Color
+    {
+        UInt8 R;
+        UInt8 G;
+        UInt8 B;
+        UInt8 A;
+    };
+
+    declare
+    {
+        // Forward declare this parameterized specialization so that it lives
+        // in TerminalCore instead of being flung to the winds of all IDL dependents.
+        interface Windows.Foundation.IReference<Microsoft.Terminal.Core.Color>;
+    }

```

Comment:
Out of curiosity, is this `declare` syntax documented somewhere? I found [Reserved keywords (MIDL 3.0)](https://docs.microsoft.com/en-us/uwp/midl-3/reserved-keywords) but no details.

Reply:
Ugh, apparently it is not. Thank you.

Reply:
From <https://github.com/MicrosoftDocs/winrt-related/issues/82#issuecomment-459158381>, it looks like complete documentation for MIDL 3.0 is not planned.